akka-cluster-typed/src/main/resources/reference.conf

```diff
@@ -13,6 +13,12 @@ akka.cluster.typed.receptionist {
   # Period task to remove actor references that are hosted by removed nodes,
   # in case of abrupt termination.
   pruning-interval = 3 s
+
+  # Shard the services over this many Distributed Data keys, with large amounts of different
+  # service keys storing all of them in the same Distributed Data entry would lead to large updates
+  # etc. instead the keys are sharded across this number of keys. This must be the same on all nodes
+  # in a cluster, changing it requires a full cluster restart (stopping all nodes before starting them again)
```

Comment:
We should add a `JoinConfigCompatChecker` for this property to ensure that all cluster members use the same value

Reply:
Added!